x-pack/legacy/plugins/alerting/server/lib/transform_action_params.ts

```diff
@@ -8,15 +8,25 @@ import Mustache from 'mustache';
 import { isString, cloneDeep } from 'lodash';
 import { AlertActionParams, State, Context } from '../types';
 
-export function transformActionParams(
-  params: AlertActionParams,
-  state: State,
-  context: Context
-): AlertActionParams {
+interface TransformActionParamsOptions {
+  alertId: string;
+  alertInstanceId: string;
+  params: AlertActionParams;
+  state: State;
+  context: Context;
+}
+
+export function transformActionParams({
+  alertId,
+  alertInstanceId,
+  context,
+  params,
+  state,
+}: TransformActionParamsOptions): AlertActionParams {
   const result = cloneDeep(params, (value: any) => {
     if (!isString(value)) return;
 
-    return Mustache.render(value, { context, state });
+    return Mustache.render(value, { alertId, alertInstanceId, context, state });
```

Comment:
I'm thinking a use case for using `alertId` and `alertInstanceId` will be for links. This isn't sufficient to create links back to Kibana. What else do we need to pass to the templates to support this?

Reply:
Created #45815 for building URLs pointing to Kibana.

Reply:
So we would need to know spaceId as well (which maybe we already do) as well as the rest of the pathing info for whatever links we're wanting to create.  Seems like we should provide functions to actions for creating those links so that the actions aren't coupled to Kibana paths.

Reply:
That's true, I haven't thought about using Mustache functions. I'm thinking the space, basePath and kibana publish host could all be handled from one function that is passed to the template. We could pass the template `baseUrl` which would be a string of all three or pass the template a function that takes a `path` as an argument and returns the full url.

Question: Should we block this PR on the URL feature or have the variables for now and implement URLs once platform has something?

Reply:
I don't think there's a huge problem providing a lot of data in the context, if we have a good feeling that we'll always be able to provide it.  And a "url function" we can add when we get it, I don't think there's any need to block this PR to wait for it.  If nothing else, having the instanceId available for diagnostic purposes when doing rando/stress testing is useful.